packages/components/src/icon/stories/index.story.tsx

```diff
@@ -47,26 +47,78 @@ FillColor.args = {
 	...Default.args,
 };
 
+/**
+ * When `icon` is a function, it will be passed the `size` prop and any other additional props.
+ */
 export const WithAFunction = Template.bind( {} );
 WithAFunction.args = {
 	...Default.args,
-	icon: () => (
-		<SVG>
-			<Path d="M5 4v3h5.5v12h3V7H19V4z" />
-		</SVG>
-	),
+	icon: ( { size }: { size?: number } ) => {
+		const doubleSize = size !== undefined ? size * 2 : size;
+
+		return (
+			<img
+				width={ doubleSize }
+				height={ doubleSize }
+				src="https://s.w.org/style/images/about/WordPress-logotype-wmark.png"
+				alt="WordPress"
+			/>
+		);
+	},
 };
+WithAFunction.parameters = {
+	docs: {
+		source: {
+			code: `
+<Icon
+  icon={ ( { size } ) => {
+    const doubleSize = size !== undefined ? size * 2 : size;
```

Comment:
I'd like to avoid manual snippets as much as possible, but I think it's unavoidable in this case.